src/subscription_manager/managercli.py

```diff
@@ -500,6 +532,20 @@ def main(self, args=None):
 
         # do the work, catch most common errors here:
         try:
+            if self.proxy_hostname or \
```

Comment:
Would a check for 'config_changed' here be sufficient (if the proxy arg/attr handling code learns to set config_changed. It currently doesn't but it seems like it should...).